### PR TITLE
KOGITO-6758: persistence-commons-postgresql: Reporting: Module re-organisation

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/KOGITO-6758

During my work for [KOGITO-6731](https://issues.redhat.com/browse/KOGITO-6731) I didn't like the smell of duplicating code.

So.. I bit the bullet and re-organised the modules to avoid all duplication.

IMO it gives a cleaner separation for "generic" and "data-index" implementations of the "reporting" mechanism. Each now only needs to add a dependency on `persistence-commons-reporting-postgresql-base` and define one class that is a concrete implementation of an interface CDI is expecting to find. Previously both "generic" and "data-index" had an overlap that I never liked and led to code duplication in the original PR (as reported by SonarCloud). 

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Review by danielezonca — APPROVED** [review]
LGTM :+1:

File: persistence-commons/persistence-commons-reporting-parent/persistence-commons-reporting-postgresql-base/src/main/java/org/kie/kogito/persistence/postgresql/reporting/bootstrap/PostgresStartupHandlerImpl.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.kie.kogito.persistence.postgresql.reporting.model.PostgresMappingDefinition;
+import org.kie.kogito.persistence.postgresql.reporting.model.PostgresMappingDefinitions;
+import org.kie.kogito.persistence.postgresql.reporting.model.PostgresPartitionField;
+import org.kie.kogito.persistence.postgresql.reporting.service.PostgresMappingServiceImpl;
+import org.kie.kogito.persistence.reporting.bootstrap.BaseStartupHandler;
+
+import io.quarkus.runtime.StartupEvent;
+
+@ApplicationScoped
+public class PostgresStartupHandlerImpl
```
Comment (MarianMacik) [inline persistence-commons/persistence-commons-reporting-parent/persistence-commons-reporting-postgresql-base/src/main/java/org/kie/kogito/persistence/postgresql/reporting/bootstrap/PostgresStartupHandlerImpl.java:36]: Where is it actually injected? Before there was `PostgresDataIndexReportingStorageService extends PostgreSqlStorageService` so it extended the storage service but now?

**Review by MarianMacik — APPROVED** [review]
Looks good, just one comment.

File: persistence-commons/persistence-commons-reporting-parent/persistence-commons-reporting-postgresql-base/src/main/java/org/kie/kogito/persistence/postgresql/reporting/bootstrap/PostgresStartupHandlerImpl.java (hunk 1 above)
Comment (manstis) [reply persistence-commons/persistence-commons-reporting-parent/persistence-commons-reporting-postgresql-base/src/main/java/org/kie/kogito/persistence/postgresql/reporting/bootstrap/PostgresStartupHandlerImpl.java:36]: Previously the specialised `StorageService` implementations (_generalised_ and _data index_) duplicated the code now in `PostgresStartupHandlerImpl` to load and setup the mappings (in `bootstrap.json`). I refactored it, removing the duplication. It does not need injection as it observes the Quarkus `StartupEvent` [here](https://github.com/kiegroup/kogito-apps/pull/1237/files/fa96cc8aa177d01e3f7be7f7915c24ea5d8d8954#diff-68d6a71388add6971b3bc638edd95f79cfdfa74378a0eb568f932086d5ac4a48R52). Therefore by including the `persistence-commons-reporting-postgresql-base` dependency you get "bootstrap" handling "for free".

Comment (MarianMacik) [reply persistence-commons/persistence-commons-reporting-parent/persistence-commons-reporting-postgresql-base/src/main/java/org/kie/kogito/persistence/postgresql/reporting/bootstrap/PostgresStartupHandlerImpl.java:36]: Right, makes sense.

**Pull request merged**